docs/source/kms.md

```diff
@@ -0,0 +1,133 @@
+# Key Management System (KMS)
+
+## What is KMS?
+
+KMS lets to manage cryptographic keys and use them to perform crypto operations. [TrustBloc KMS][trustbloc-kms] is a
+server implementation of [KMS][kms-api] and [Crypto][crypto-api] APIs from the [Aries Framework Go][aries-framework-go].
+
+It can be used as a [remote KMS][remote-kms] and a [remote Crypto][remote-crypto] for the framework's `webkms`
+implementation.
+
+## Running KMS server
+
+KMS server can be run as a standalone binary or a docker container. Refer [here][running-kms-server] for the instructions
+on how to run a server and supported startup flags and options.
+
+## KMS in a TrustBloc environment
+
+KMS server can play a different role depending on the concrete setup.
+
+In the TrustBloc environment, the KMS server is used as an authorization KMS for supporting [ZCAP-LD][zcap-ld] scheme
+and as an operational KMS for regular user's crypto operations.
+
+### Authorization KMS
+
+Authorization KMS allows creating Controller identity (identified by a cryptographic key pair) and then using it for
+signing requests in a ZCAP-LD authorization model.
+
+The key store for the Controller uses the server's database for storing keys. These keys are encrypted with a
+[Shamir secret lock][shamir-secret-lock].
+
+To unlock a key store, the controller's secret share needs to be present in a `Secret-Share` header for every request.
+The other share comes from the Auth server. The endpoint for fetching that share is configured with `KMS_AUTH_SERVER_URL`
+environment variable.
+
+[Server Key Manager][kms-architecture] is protected with a [Local secret lock][local-secret-lock].
+
+Example configuration options for the Authorization KMS in `docker-compose`:
+
+```yaml
+authz-kms.trustbloc.local:
```

Comment:
why docker compose example

Reply:
Other examples from bloc-docs use docker-compose. Do you suggest using examples with flags, or maybe both (cli with flags and docker-compose with env variables)? cc @fqutishat